android-components/components/lib/crash/src/main/java/mozilla/components/lib/crash/service/GleanCrashReporterService.kt

```diff
@@ -121,64 +178,21 @@ class GleanCrashReporterService(
+                DecodeSequenceMode.WHITESPACE_SEPARATED,
+            )
+            for (action in actionSequence) {
+                action.submit()
+            }
         } catch (e: IOException) {
             logger.error("Error reading crash file", e)
             return
-        }
-
-        var uncaughtExceptionCount = 0
-        var caughtExceptionCount = 0
-        var mainProcessNativeCodeCrashCount = 0
-        var foregroundChildProcessNativeCodeCrashCount = 0
-        var backgroundChildProcessNativeCodeCrashCount = 0
-        // These keys are deprecated and should be removed after a period to allow for persisted
-        // crashes to be submitted.
-        var fatalNativeCodeCrashCount = 0
-        var nonfatalNativeCodeCrashCount = 0
-
-        // It's possible that there was more than one crash recorded in the file so process each
-        // line and accumulate the crash counts.
-        lines.forEach { line ->
-            when (line) {
-                UNCAUGHT_EXCEPTION_KEY -> ++uncaughtExceptionCount
-                CAUGHT_EXCEPTION_KEY -> ++caughtExceptionCount
-                MAIN_PROCESS_NATIVE_CODE_CRASH_KEY -> ++mainProcessNativeCodeCrashCount
-                FOREGROUND_CHILD_PROCESS_NATIVE_CODE_CRASH_KEY -> ++foregroundChildProcessNativeCodeCrashCount
-                BACKGROUND_CHILD_PROCESS_NATIVE_CODE_CRASH_KEY -> ++backgroundChildProcessNativeCodeCrashCount
-                FATAL_NATIVE_CODE_CRASH_KEY -> ++fatalNativeCodeCrashCount
-                NONFATAL_NATIVE_CODE_CRASH_KEY -> ++nonfatalNativeCodeCrashCount
-            }
-        }
-
-        // Now, record the crash counts into Glean
-        if (uncaughtExceptionCount > 0) {
-            CrashMetrics.crashCount[UNCAUGHT_EXCEPTION_KEY].add(uncaughtExceptionCount)
-        }
-        if (caughtExceptionCount > 0) {
-            CrashMetrics.crashCount[CAUGHT_EXCEPTION_KEY].add(caughtExceptionCount)
-        }
-        if (mainProcessNativeCodeCrashCount > 0) {
-            CrashMetrics.crashCount[MAIN_PROCESS_NATIVE_CODE_CRASH_KEY].add(
-                mainProcessNativeCodeCrashCount,
-            )
-        }
-        if (foregroundChildProcessNativeCodeCrashCount > 0) {
-            CrashMetrics.crashCount[FOREGROUND_CHILD_PROCESS_NATIVE_CODE_CRASH_KEY].add(
-                foregroundChildProcessNativeCodeCrashCount,
-            )
-        }
-        if (backgroundChildProcessNativeCodeCrashCount > 0) {
-            CrashMetrics.crashCount[BACKGROUND_CHILD_PROCESS_NATIVE_CODE_CRASH_KEY].add(
-                backgroundChildProcessNativeCodeCrashCount,
-            )
-        }
-        if (fatalNativeCodeCrashCount > 0) {
-            CrashMetrics.crashCount[FATAL_NATIVE_CODE_CRASH_KEY].add(fatalNativeCodeCrashCount)
-        }
-        if (nonfatalNativeCodeCrashCount > 0) {
-            CrashMetrics.crashCount[NONFATAL_NATIVE_CODE_CRASH_KEY].add(nonfatalNativeCodeCrashCount)
+        } catch (e: SerializationException) {
```

Comment:
Any existing non-submitted crash files will cause this exception here, but probably not worth handling. We will just miss those crashes. That would happen when the app crashes and an update is applied to a version that has this change, before the app is restarted (and submits the crash). Then we couldn't parse the old file format anymore. That should be a (very) small % of clients though. Worth pointing out in case we see a drop while this rolls out.

Reply:
Yep, I acknowledge that. I didn't think it would be that high a priority (due to low likelihood) to make sure both formats are supported.

Reply:
Yeah, I agree we can live with that. Mostly commenting for documentation purposes.